components/push/src/subscriber.rs

```diff
@@ -13,10 +13,56 @@ use crate::config::PushConfiguration;
 use crate::crypto::{Crypto, Cryptography, KeyV1 as Key};
 use crate::error::{self, ErrorKind, Result};
 use crate::storage::{PushRecord, Storage, Store};
+// TODO(teshaq): those will be replaced with rust structs in the next
+// uniffication step
+use crate::msg_types::{
+    DispatchInfo, KeyInfo, PushSubscriptionChanged, SubscriptionInfo, SubscriptionResponse,
+};
 
 const UPDATE_RATE_LIMITER_INTERVAL: u64 = 24 * 60 * 60; // 500 calls per 24 hours.
 const UPDATE_RATE_LIMITER_MAX_CALLS: u16 = 500;
 
+impl From<(RegisterResponse, Key)> for SubscriptionResponse {
```

Comment:
I'm not too happy with this... 

while the `From` trait is really cool, we have too many types here, like do we really need a `KeyInfo` and a `Key` type when they basically return the same thing. 

Will need to think about this a bit and I might create a follow-up to clean this up after we are done uniffing - In the mean time, in the next PR I'll be doing a direct translation, simply replacing the `protobuf` types with rust types